Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Upsert.ts

```diff
@@ -3,64 +3,82 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
-import { Collection, ClientSession, MongoClient, WithId, FindOptions } from 'mongodb';
-import { UpsertResult, UpsertRequest, documentIdForSuperclassInfo, BlockingDocument } from '@edfi/meadowlark-core';
+/* eslint-disable no-underscore-dangle */
+
+import { Collection, ClientSession, MongoClient, WithId } from 'mongodb';
+import {
+  UpsertResult,
+  UpsertRequest,
+  documentIdForSuperclassInfo,
+  BlockingDocument,
+  DocumentUuid,
+} from '@edfi/meadowlark-core';
 import { Logger } from '@edfi/meadowlark-utilities';
 import { MeadowlarkDocument, meadowlarkDocumentFrom } from '../model/MeadowlarkDocument';
-import { getDocumentCollection, writeLockReferencedDocuments, asUpsert } from './Db';
+import { writeLockReferencedDocuments, asUpsert, limitFive, getDocumentCollection, onlyReturnDocumentUuid } from './Db';
 import { onlyDocumentsReferencing, validateReferences } from './ReferenceValidation';
 
-// MongoDB FindOption to return at most 5 documents
-const limitFive = (session: ClientSession): FindOptions => ({ limit: 5, session });
-
 const moduleName: string = 'mongodb.repository.Upsert';
+
 export async function upsertDocumentTransaction(
-  { resourceInfo, documentInfo, documentUuidInserted, meadowlarkId, edfiDoc, validate, traceId, security }: UpsertRequest,
-  client: MongoClient,
+  {
+    resourceInfo,
+    documentInfo,
+    documentUuidForInsert,
```

Comment:
There are two different documentUuids at play here. This one is provided for an insert situation, either speculatively by the meadowlark-core handler, or by updateByDocumentId when it's an insert for identity change reasons.